File: OneDrive/Desktop/program1/fibo_bench.cpp
#include <iostream>
#include <fstream>
#include <chrono>
using namespace std;
using namespace std::chrono;
long long fibo_recursive(int n){return n<=1?1:fibo_recursive(n-1)+fibo_recursive(n-2);}
long long fibo_iterative(int n){if(n<=1)return 1;long long a=1,b=1;for(int i=2;i<=n;i++){long long c=a+b;a=b;b=c;}return b;}
long long time_rec(int n){auto t0=high_resolution_clock::now();volatile long long v=fibo_recursive(n);auto t1=high_resolution_clock::now();return duration_cast<milliseconds>(t1-t0).count();}
long long time_it (int n){auto t0=high_resolution_clock::now();volatile long long v=fibo_iterative(n);auto t1=high_resolution_clock::now();return duration_cast<milliseconds>(t1-t0).count();}
int main(){
  ofstream csv("fibo_benchmark_cpp.csv");csv<<"n,fib,rec_ms,it_ms\n";
  for(int n=0;n<=30;n++){long long val=fibo_iterative(n);
    long long rec=0,it=0;for(int t=0;t<3;t++){rec+=time_rec(n);it+=time_it(n);}
    csv<<n<<","<<val<<","<<(rec/(double)3)<<","<<(it/(double)3)<<"\n";
    if(n>=40&&n<=50) cout<<"fibo("<<n<<")="<<val<<"\n";}
}

File: OneDrive/Desktop/program1/fibo_bench.py
import time,csv
def fibo_recursive(n): return 1 if n<=1 else fibo_recursive(n-1)+fibo_recursive(n-2)
def fibo_iterative(n):
    if n<=1: return 1
    a,b=1,1
    for _ in range(2,n+1): a,b=b,a+b
    return b
def time_ms(fn,n,trials=3):
    total=0.0
    for _ in range(trials):
        t0=time.perf_counter();fn(n);t1=time.perf_counter()
        total+=(t1-t0)*1000.0
    return total/trials
with open("fibo_benchmark_python.csv","w",newline="") as f:
    w=csv.writer(f);w.writerow(["n","fib","rec_ms","it_ms"])
    for n in range(30+1):
        val=fibo_iterative(n)
        rec=time_ms(fibo_recursive,n,3)
        it=time_ms(fibo_iterative,n,3)
        w.writerow([n,val,rec,it])
        if 40<=n<=50: print(f"fibo({n})={val}")
print("Wrote fibo_benchmark_python.csv")
